src/components/Facility/PatientNotesList.tsx

```diff
@@ -23,57 +24,59 @@ interface PatientNotesProps {
   setReplyTo?: (reply_to: PatientNotesModel | undefined) => void;
 }
 
-const pageSize = RESULTS_PER_PAGE_LIMIT;
-
 const PatientNotesList = (props: PatientNotesProps) => {
-  const { state, setState, reload, setReload, thread, setReplyTo } = props;
-
-  const [isLoading, setIsLoading] = useState(true);
+  const { state, setState, thread, setReplyTo, setReload, patientId, reload } =
+    props;
+  const queryClient = useQueryClient();
 
-  const fetchNotes = async () => {
-    setIsLoading(true);
-    const { data }: any = await request(routes.getPatientNotes, {
-      pathParams: { patientId: props.patientId },
-      query: {
-        offset: (state.cPage - 1) * RESULTS_PER_PAGE_LIMIT,
-        thread,
-      },
-    });
+  const { data, isLoading, fetchNextPage, hasNextPage } = useInfiniteQuery({
+    queryKey: ["notes"],
```

Comment:
add thread, patientId and consultationId (if present) to the query key.

refer: https://tkdodo.eu/blog/effective-react-query-keys